networkit/cpp/centrality/test/CentralityGTest.cpp

```diff
@@ -1293,6 +1293,37 @@ TEST_P(CentralityGTest, testTopCloseness) {
     }
 }
 
+TEST_F(CentralityGTest, testTopClosenessWithNodeList) {
+    METISGraphReader reader;
+    Graph G = reader.read("input/lesmis.graph");
+    constexpr count k = 10;
+    const std::vector<node> nodeList{0, 1, 2, 3, 4, 5, 11, 26, 48, 64};
+
+    // We expect complete TopCloseness to not contain nodes 0-5, while
+    // restricted should. The first element for both complete and
+    // restricted TopCC should be 11. {26, 48, 64} should also be present
+    // in both results (but position might differ).
+    for (auto firstHeu : {true, false}) {
+        for (auto secHeu : {true, false}) {
+            TopCloseness topC(G, k, firstHeu, secHeu);
+            topC.run();
+            auto topCNodes = topC.topkNodesList();
+            TopCloseness topCR(G, k, firstHeu, secHeu, nodeList);
+            topCR.run();
+            auto topCRNodes = topCR.topkNodesList();
+            EXPECT_EQ(topCNodes[0], topCRNodes[0]);
+            for (node i : {0, 1, 2, 3, 4, 5}) {
+                EXPECT_EQ(std::find(topCNodes.begin(), topCNodes.end(), i), topCNodes.end());
```

Comment:
This can be simplified a bit using something like `EXPECT_THAT(topCNodes, Contains(i))`. See here for more: https://github.com/google/googletest/blob/main/docs/reference/matchers.md

Reply:
You can also avoid the loop by using `EXPECT_THAT(topCNodes, UnorderedElementsAre({0, 1, ...}));`. The link above gives you a wide range of container matchers.

Reply:
Thanks for the review ... wasn't aware of gmock-based tests. Don't know if that makes it really simpler though ;) Had to use `IsSuperSetOf`, since `UnorderedElementsAre` (or similar) require the same number of elements from the argument and the matcher.

Reply:
Ah true, then `IsSuperSetOf` is what you need. Overall I think that now it's easier to understand what you are testing than having all those `std::find`.

Reply:
Btw, you are testing that some nodes are in the top-k ranking. Shouldn't you also test the ranking?

Reply:
Hm, you mean what we expect as ranking vector (when entering a `nodeList`)?

Reply:
As far as I understand, `topkNodesList` returns the top-k nodes sorted by non-increasing closeness score. If that's the case, then we should test it.

Reply:
Ah, right! Added tests for ensuring proper rank-sorting.